src/main/kotlin/org/openrewrite/kotlin/KotlinTypeMapping.kt

```diff
@@ -546,7 +550,7 @@ class KotlinTypeMapping(typeCache: JavaTypeCache, firSession: FirSession) : Java
                 }
             }
             if (resolvedDeclaringType == null) {
-                return null
+                resolvedDeclaringType = TypeUtils.asFullyQualified(type(firFileSymbol.fir))
```

Comment:
@knutwannheden I needed to figure out where we should fall back to the FirFileSymbol. This "could" technically set the incorrect owner if we don't assign the owner appropriately, but it also isolates incorrect types to each file.

Curious, what do you think about falling back onto the FirFileSymbol by default?

Reply:
> Curious, what do you think about falling back onto the FirFileSymbol by default?

I think that sounds reasonable. I wanted to do something similar myself, but first wanted to discuss that with you. So instead I added this owner stack, which is easy enough to remove again.